### DISCUSSION TITLE
Building with Cycles OneAPI

### DISCUSSION THREAD

**Discussion by Draise14** [Bforartists/Bforartists#3152] [General]
![image](https://user-images.githubusercontent.com/35357409/179651938-e1d9df4c-ea30-4e79-a598-b370b786017c.png)

Are devbuilds supposed to compile this? If so... should we document these dependencies?

Some clues:
https://developer.blender.org/T96840

Error I get on a "Release" build. 
![image](https://user-images.githubusercontent.com/35357409/179651993-44bee50e-4847-4d78-86c7-e819421064d2.png)

**Comment by ReinerBforartists** [comment]
We should offer what the release version of Blender offers. So does the release version come with the api?

It would be interesting to know if Blender has the same linker error. This would be worth a bug report then.

**Reply by Draise14** [reply]
I did a vanilla blender compile, same error. So it's there too.

**Comment by ghost** [comment]
I think I got it now, we need to follow the steps here and install that other compiler if we want to ship with oneapi cycles, that said not sure if Blender ships with it yet, but on my machine compiling a release doesn't enable oneapi cycles device, will double check

https://developer.blender.org/T96840

**Comment by ghost** [comment]
And yes the compiler link in the link above is not direct, you have to search for "offline compiler" on that intel page when you go into it

It will give you a list of zip files of different versions of the compiler,

Will grab a direct link

**Comment by ghost** [comment]
here
https://www.intel.com/content/www/us/en/developer/articles/tool/oneapi-standalone-components.html#:~:text=Mar.%2029%2C%202022-,Intel%C2%AE%20Graphics%20Offline%20Compiler%20for%20OpenCL%E2%84%A2%20Code,-Reduce%20runtime%20overhead

**Comment by ghost** [comment]
![image](https://user-images.githubusercontent.com/25552173/179685540-c04bcbd7-d49d-45ab-b4cc-4bc95432eba7.png)

only step one is covered by lib folder, previously I thought only the lib folder was enough

so this means that we only need to modify the first 3 and last 1 of the cmake options in the above image

because the others are already detected from the lib folder

**Comment by ghost** [comment]
confirming binaries are not enabled with release config
only device
![image](https://user-images.githubusercontent.com/25552173/179688132-ec20cf1d-6a62-4990-904d-c91d80517f25.png)

**Comment by ReinerBforartists** [comment]
Sheesh ...

**Comment by ReinerBforartists** [comment]
Linux comes still without. But Mac and Windows seems to be shipped with this now.
![release](https://user-images.githubusercontent.com/13919702/179698578-26bb7757-5843-47b7-9f6b-64f2ce8ced18.jpg)

**Comment by ReinerBforartists** [comment]
Hm, compiled just fine without further downloads here. Seems that everything required is already onboard.

**Reply by Draise14** [reply]
Wait, you got no error?

**Comment by ReinerBforartists** [comment]
But wait, do we need to compile the binaries now too or not?

**Comment by ghost** [comment]
Yes no need to enable one api binaries

**Comment by Draise14** [comment]
I had a brief chat, this is still a highly experimental feature and closer to 3.3 release the compilation with this will be easier. But for now it is not that automatic for the binaries. But it SHOULD just work with the device.

**Comment by Draise14** [comment]
Looks like the oneAPI binaries on Windows was enabled in this weeks merge. Will have to test.

**Comment by Draise14** [comment]
There is a bit of an update to get OneAPI binaries and dependencies involved. They updated the documentation. 
[Link to Documentation](https://wiki.blender.org/wiki/Building_Blender/GPU_Binaries#Intel)

**Comment by ReinerBforartists** [comment]
Thanks, will have a closer look :)

EDIT, Um ...

> The HIP SDK for building AMD GPU binaries is not currently publicly available, we hope this will be released soon.

So nothing has changed ^^

**Reply by ghost** [reply]
But they sent us one uploaded 9n next loud?
@Draise14

**Reply by Draise14** [reply]
Yeah, the HIP libraries we have on nextcloud, I may ask for an update and upload if I get it - but the OneAPI is now open and a seperate one to this thread.
